internal/services/formatters/php/phpcs/formatter_test.go

```diff
@@ -36,17 +39,52 @@ func TestStartCFlawfinder(t *testing.T) {
 		config := &cliConfig.Config{}
 		config.WorkDir = &workdir.WorkDir{}
 
-		output := "{ \"files\":{ \"\\/src\\/XSS\\/XSS_level5.php\":{ \"errors\":1, \"warnings\":4, \"messages\":[ { \"message\":\"User input detetected with $_SERVER.\", \"source\":\"PHPCS_SecurityAudit.Drupal7.UserInputWatch.D7UserInWaWarn\", \"severity\":5, \"fixable\":false, \"type\":\"WARNING\", \"line\":14, \"column\":39 }, { \"message\":\"Easy XSS detected because of direct user input with $_SERVER on echo\", \"source\":\"PHPCS_SecurityAudit.BadFunctions.EasyXSS.EasyXSSerr\", \"severity\":5, \"fixable\":false, \"type\":\"ERROR\", \"line\":14, \"column\":39 } ] } } }"
+		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return(outputAnalysis, nil)
 
-		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return(output, nil)
+		service := formatters.NewFormatterService(analysis, dockerAPIControllerMock, config)
+		formatter := NewFormatter(service)
+
+		formatter.StartAnalysis("")
+
+		totalBySeverity := map[severities.Severity]int{}
+		for _, v := range analysis.AnalysisVulnerabilities {
+			totalBySeverity[v.Vulnerability.Severity]++
+		}
+		assert.Equal(t, 0, totalBySeverity[severities.Unknown])
+		assert.Equal(t, 96, totalBySeverity[severities.Info])
+		assert.Equal(t, 23, totalBySeverity[severities.Critical])
+		assert.Equal(t, 0, totalBySeverity[severities.Low])
+		assert.Equal(t, 0, totalBySeverity[severities.Medium])
+		assert.Equal(t, 0, totalBySeverity[severities.High])
+	})
+	t.Run("should success execute container and validate if only problems of type warning, error are added in vulnerability struct", func(t *testing.T) {
+		dockerAPIControllerMock := testutil.NewDockerMock()
+		analysis := &entitiesAnalysis.Analysis{}
+		config := &cliConfig.Config{}
+		config.WorkDir = &workdir.WorkDir{}
+
+		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return(outputAnalysis, nil)
 
 		service := formatters.NewFormatterService(analysis, dockerAPIControllerMock, config)
 		formatter := NewFormatter(service)
 
 		formatter.StartAnalysis("")
 
+		totalErrorsFoundInOutput := 0
+		var outputParsed map[string]interface{}
+		assert.NoError(t, json.Unmarshal([]byte(outputAnalysis), &outputParsed))
+		for _, r := range outputParsed["files"].(map[string]interface{}) {
+			var newR *entities.Result
+
+			bytes, _ := json.Marshal(r)
+			_ = json.Unmarshal(bytes, &newR)
+			totalErrorsFoundInOutput += len(newR.Messages)
+		}
+
+		assert.Equal(t, 120, totalErrorsFoundInOutput, "Validation: \"This implies that some PHP code is not scanned by PHPCS\" not was been applied ")
```

Comment:
This assert make no sense to me, why not  assert the `analysis.AnalysisVulnerabilities` contents?

Reply:
Note the title of the test we want to know if only `error` and `warning` problems are added in the analysis object, so here it was just a validation to compare if 120 vulnerabilities were actually identified in the output and then there is another one assert to check if it was filtered and added only 119 in the final analysis.

Reply:
Yeah, but this make no sense to me. Basically here we are testing the json.Unmarshal. We already known that the mock output contains the 120 vulnerabilities, I don't think that we need test this here. Just assert that the correctly vulnerabilities was added on Analysis object would be enough.

Reply:
for me yes, is necessary check if was ignored with success and validated before and after of the results, what does think @nathanmartinszup @lucasbrunozup @iancardosozup

Reply:
So, we are testing the testing here, I disagree with this. Also I think that we should test the formatter behavior, not that json.Marshal/json.Unmarhsal fill all the struct values. 

Why we would even change this mock output? I think that the only case when we need to change this output is when the phpcs change the output format. 

If we still want to test this I think that we should write this test in a better way.

Reply:
I agree, what would be the best way for you?

Reply:
I think that would be better to create another test (separated from TestStartCFlawfinder) with just this assert, for example:

```go
func TestTotalVulnerabilitiesFromCodeSnifferOutput(t *testing.T) {
	totalErrorsFoundInOutput := 0

	var outputParsed map[string]interface{}
	assert.NoError(t, json.Unmarshal([]byte(outputAnalysis), &outputParsed))

	for _, r := range outputParsed["files"].(map[string]interface{}) {
		var newR *entities.Result

		bytes, _ := json.Marshal(r)
		_ = json.Unmarshal(bytes, &newR)
		totalErrorsFoundInOutput += len(newR.Messages)
	}

	assert.Equal(t, 120, totalErrorsFoundInOutput, "Validation: \"This implies that some PHP code is not scanned by PHPCS\" not was been applied ")
}
```

But as I say, I don't think that we need this test. Let's see what @nathanmartinszup and @iancardosozup think about this test before we move on.

One more point why I think that this test is not necessary/good: I don't think that we should expose the CodeSniffer structs that represents the output schema as a public API. As was done on #925, #910, #902, #901 and other PRs that fix the issue #718  all structs declared on `entities` packages was moved to their respective formatter packages and changed to be private  (there is no needed for them be public) and as I say [here](https://github.com/ZupIT/horusec/pull/838#discussion_r761959089) I don't think that test private functions/structs is a good strategy at all.

BTW the test func should be renamed from `TestStartCFlawfinder` to `TestStartPHPCodeSniffer`.

Reply:
Another way to to test this is creating another output mock (just for this test), with just the results that should be ignored by formatter, and them we assert that `analysis.AnalysisVulnerabilities` is empty. What do you think about this approach?

Reply:
I think it's the best idea :)